float32[10] data
